src/components/Activity/ActivityList.js

```diff
@@ -25,99 +39,123 @@ const ActivityList = React.memo(({ apps, activities, clearActivity }) => {
     [activities, apps]
   )
 
-  const handleDiscard = useCallback(
-    ({ transactionHash }) => {
-      clearActivity(transactionHash)
-    },
-    [clearActivity]
+  const canClear = useMemo(
+    () =>
+      activityItems.some(({ status }) => status !== ACTIVITY_STATUS_PENDING),
+    [activityItems]
+  )
+  const maxHeight = Math.min(
+    MAX_LIST_HEIGHT_CLAMP,
+    Math.ceil(height - MIN_LIST_HEIGHT_ADJUST)
   )
-
-  const transitionKeys = activity => activity.transactionHash
 
   return (
-    <div css="position: relative">
-      <Transition
-        native
-        items={activityItems.length === 0}
-        from={{ opacity: 0 }}
-        enter={{ opacity: 1 }}
-        leave={{ opacity: 0 }}
-        config={springs.lazy}
+    <div
+      css={`
+        width: ${42 * GU}px;
+      `}
+    >
+      <div
+        css={`
+          display: flex;
+          align-items: center;
+          justify-content: space-between;
+          height: ${4 * GU}px;
+          padding: 0 ${2 * GU}px;
+          border-bottom: 1px solid ${theme.border};
+        `}
       >
-        {show =>
-          show &&
-          (transitionStyles => (
-            <div
-              css={`
-                position: absolute;
-                top: 0;
-                width: 100%;
-                display: flex;
-                flex-direction: column;
-                align-items: center;
-                padding: ${cssgu`8gu 4gu 8gu`};
-              `}
-              style={transitionStyles}
-            >
-              <IconEmptyState />
-              <p
+        <label
+          css={`
+            ${textStyle('label2')}
+            color: ${theme.surfaceContentSecondary};
+          `}
+        >
+          Activity
+        </label>
+        {canClear && (
+          <ButtonText
+            onClick={clearActivities}
+            css={`
+              padding: 0;
+              ${textStyle('label2')}
+            `}
+          >
+            Clear all
+          </ButtonText>
+        )}
+      </div>
+      <div
+        css={`
+          max-height: ${maxHeight}px;
+          overflow-x: hidden;
+          overflow-y: scroll;
```

Comment:
Should we use `auto` so that it doesn’t appear until it’s needed? The scrollbar blinks on Firefox while appearing, so it would also solve this :-)

```suggestion
          overflow-y: auto;
```